src/backend/expungeservice/endpoints/users.py

```diff
@@ -2,24 +2,49 @@
 from flask import request, abort, jsonify
 from werkzeug.security import generate_password_hash
 
-# TODO temp hack - replace with table from DB
-users = {}
+from flask import g
+from expungeservice.database.user import create_user, get_user_by_email
+from expungeservice.endpoints.auth import admin_auth_required
+from expungeservice.request import check_data_fields
+from psycopg2.errors import UniqueViolation
+
 
 class Users(MethodView):
+    @admin_auth_required
     def post(self):
+        """
+        Create a new user with provided email, password, and admin flag.
+        - If required fields are missing in the request, return 400
+        - Password must be 8 or more characters long. Otherwise return 422
+        - Email must not already be in use by an existing user. Otherwise return 422
+        If success, return 201 with the new user's email, admin flag, and creation timestamp.
+        """
+
         data = request.get_json()
 
         if data == None:
             abort(400)
```

Comment:
@maxkwallace Aside question: I followed the pre-existing pattern that most endpoint errors are returned as a string and status code, but this one written before me is being raised instead. I also am using abort(400) in check_data_fields() so the error will propagate. Do you have thoughts on raising vs returning errors generally?

Reply:
Nope, I don't know enough about the conventions in Flask to comment offhand.